src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -86,14 +87,29 @@ private BoundExpression LowerSwitchExpression(BoundSwitchExpression node)
                 if (node.DefaultLabel != null)
                 {
                     result.Add(_factory.Label(node.DefaultLabel));
-                    // https://github.com/dotnet/roslyn/issues/27747 Need a dedicated platform exception type to throw for input not matched.
-                    result.Add(_factory.ThrowNull());
+                    var objectType = _factory.SpecialType(SpecialType.System_Object);
+                    var thrownExpression =
+                        (optionalSavedInput != null &&
+                                implicitConversionExists(optionalSavedInput, objectType) &&
+                                _factory.WellKnownMember(WellKnownMember.System_MatchFailureException__ctor1, isOptional: true) is MethodSymbol newMatchFailureException1)
+                            ? _factory.New(newMatchFailureException1, _factory.Convert(objectType, optionalSavedInput)) :
+                        (_factory.WellKnownMember(WellKnownMember.System_MatchFailureException__ctor0, isOptional: true) is MethodSymbol newMatchFailureException0)
+                            ? _factory.New(newMatchFailureException0) :
+                        _factory.New(_factory.WellKnownMethod(WellKnownMember.System_InvalidOperationException__ctor));
```

Comment:
I realize that throwing `InvalidOperationException` is an edge case that's probably not going to happen very often, but shouldn't the thrown exception have a better message than the default "Operation is not valid due to the current state of the object."? The message could probably be the same as what `MatchFailureException.Message` is going to be.

Reply:
I do not expect any message for `InvalidOperationException .Message`. The use of this exception is merely to allow using pattern-matching on platforms that precede the introduction of `MatchFailureException`. The message in `MatchFailureException.Message` will depend on your `corefx` implementation, none of which exist yet.